docs/coalton-lisp-interop.md

```diff
@@ -0,0 +1,132 @@
+# Coalton-Lisp Interoperation
+
+## Introduction
+
+Coalton is a language embedded in Lisp, and indeed, Coalton compiles to Lisp code. This document describes what Coalton promises for interoperation.
+
+## Interaction Mode
+
+First and foremost, there are two ways to globally compile Coalton. This is determined by an environment variable `COALTON_ENV` which in turns sets a global variable `*interaction-mode*`. There are two options for this variable:
+
+- `COALTON_ENV=development`, which sets `*interaction-mode*` to `:development`
+- `COALTON_ENV=release`, which sets `*interaction-mode*` to `:release`
+
+As the names imply, these two modes have different behaviors that optimize for either development or release.
+
+**WARNING**: *All* Coalton code, *including* the standard library, *must* be compiled in the same mode. We advise application writers to set the environment variable during their application's build process before Coalton is compiled, and to ensure caches are not stale.
```

Comment:
Maybe we should provide some way to either allow interop between modes or to tag definitions with their mode and error when new code is compiled using a different mode

Reply:
> tag definitions with their mode and error when new code is compiled using a different mode

I like this. I think we could have a header in the generated code that checks the mode instead of tagging each definition

Reply:
Added issue #206